webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.test.js

```diff
@@ -1,42 +1,60 @@
 import React from 'react';
-import ForemanModal from '.';
+import PropTypes from 'prop-types';
+import ForemanModal from './ForemanModal';
+import ForemanModalHeader from './subcomponents/ForemanModalHeader';
+import ForemanModalFooter from './subcomponents/ForemanModalFooter';
 import { testComponentSnapshotsWithFixtures } from '../../common/testHelpers';
 
-const headerChild = (
-  <ForemanModal.Header>this is the header</ForemanModal.Header>
-);
-const footerChild = (
-  <ForemanModal.Footer>this is the footer</ForemanModal.Footer>
-);
+// This file is for unit tests of the ForemanModal component
+
+const headerChild = <ForemanModalHeader>this is the header</ForemanModalHeader>;
+const footerChild = <ForemanModalFooter>this is the footer</ForemanModalFooter>;
 const modalBody = <div>This is the body</div>;
+const openState = { open: true };
+const closedState = { open: false };
+const onClose = jest.fn();
 
 const fixtures = {
   'should render': {
-    isOpen: true,
+    id: 'render',
     title: 'Test modal',
-    onClose: jest.fn(),
+    openState,
+    onClose,
   },
   'should render closed': {
-    isOpen: false,
+    id: 'closed',
     title: 'Test modal',
-    onClose: jest.fn(),
+    openState: closedState,
+    onClose,
   },
   'renders when header and footer are supplied': {
-    isOpen: true,
+    id: 'withSuppliedChildren',
     title: 'Test modal',
-    onClose: jest.fn(),
     children: [headerChild, modalBody, footerChild],
-  },
-  'renders header and footer in correct order regardless of ordering of children': {
-    isOpen: true,
-    title: 'Test modal',
-    onClose: jest.fn(),
-    children: [modalBody, footerChild, headerChild],
+    openState,
+    onClose,
   },
 };
 
 describe('ForemanModal', () => {
   describe('rendering', () => {
     testComponentSnapshotsWithFixtures(ForemanModal, fixtures);
   });
+  describe('PropTypes', () => {
+    it('requires an id prop', () => {
```

Comment:
@sharvit In general I totally agree that PropTypes don't need their own tests.  In this case I just thought that since `id` is such an important part of `ForemanModal`, and defines one of the essential features of the implementation, that it would be good to state that in test form.  I'm fine deleting the test if you still think it's unnecessary, but just wanted to explain my reason :)

Reply:
@sharvit After adding the integration test I changed my mind.  Removing.